### PR TITLE
endpoint for registering mentor

### LINKED ISSUE #216: API: create an endpoint for registering mentor
The route should be `/api/user/register/mentor`, method POST and the body will contain
- `name` (string)
- `organizationId` (number).

### PR DESCRIPTION
**Pull request by nuha90**
# Description

create an endpoint for registering mentor

Fixes #216 

# How to test?

swagger and postman

# Checklist

- [x] I have performed a self-review of my own code

### CODE REVIEW

File: src/server/api/routes/mentor.router.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ *            - name
+ *            - organization_id
+ *          properties:
+ *            name:
+ *              type: string
+ *            organization_id:
+ *              type: integer
+ *    responses:
+ *      201:
+ *        description: Event created
```
Comment (tpetrina) [inline src/server/api/routes/mentor.router.js:32]: Should be `Mentor registered`.

**Review by tpetrina — CHANGES_REQUESTED** [review]
Looks good, but I just realized you are missing a small piece. 
You need to also create a new entry in`user_roles` for the newly registered mentor. The role you should link against is mentor. See @AkoAyukngat and his PR, you both need to do similar thing.

File: src/server/api/controllers/mentor.controller.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const knex = require('../../config/db');
+
+const registerMentor = async (body) => {
+  const insertNewMentor = await knex('users').insert({
```
Comment (tpetrina) [inline src/server/api/controllers/mentor.controller.js:4]: This should be `newMentorId`.

File: src/server/api/controllers/mentor.controller.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    name: body.name,
+    organization_id: body.organizationId,
+    slack_id: body.slackId,
+  });
+  const getMentorId = await knex('roles').where('name', 'mentor').first('id');
```
Comment (tpetrina) [inline src/server/api/controllers/mentor.controller.js:9]: This should be named `mentorRole`.

**Review by tpetrina — DISMISSED** [review]

**Review by tpetrina — APPROVED** [review]
Nice!

**Review by claudion3 — APPROVED** [review]

**Pull request merged**